src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -289,6 +285,91 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
     }
   }
 
+  /**
+    * Collect data from Blocks Table, Operations Table, and the Tezos RPC,
+    * return the appropriate list of VoteAggregates, and write them to the Votes table.
+    * @param ex the needed ExecutionContext to combine multiple database operations
+    * @return a future result of the number of rows stored to db, if supported by the driver
+    */
+  private[this] def processTezosVotes()(implicit ex: ExecutionContext): Future[Option[Int]] = {
+    import cats.instances.list._
+    import cats.instances.future._
+    import cats.syntax.traverse._
+    import cats.implicits._
+    import slick.jdbc.PostgresProfile.api._
+
+    logger.info("Processing latest Tezos votes data...")
+
+    def partitionByVote(fields: Seq[VotingFields]): (Seq[VotingFields], Seq[VotingFields], Seq[VotingFields]) = {
+      val (yays, notYays) = fields.partition(x => x._6 == Some("yay"))
+      val (nays, notYaysOrNays) = notYays.partition(x => x._6 == Some("nay"))
+      val (passes, _) = notYaysOrNays.partition(x => x._6 == Some("pass"))
+      (yays, nays, passes)
+    }
+
+    def calculateStake(fields: Seq[VotingFields]): Future[BigDecimal] =
+      Future
+        .traverse(
+          fields.map {
+            case (_, _, hash, _, _, _, address) =>
+              (BlockHash(hash), AccountId(address.get))
+          }
+        ) { x =>
+          tezosNodeOperator.getAccountBalanceForBlock(x._1, x._2)
```

Comment:
Stake should be derived from the `bakers` table, not the `accounts` table. 'Stake` has a very particular meaning in Tezos and the correct way to pull it out is using baker information.

@anonymoussprocket

Reply:
@anonymoussprocket and I discussed that there was no easy way to get account balances as of block level from the current database schema, so we hit the tezos node in this function, not the database. 

I'm not a fan of this approach actually, if querying the baker's table (or the db in general) would solve this issue, I'd be okay with it.

Reply:
Oh, I misread this, do you mean the baker rpc for tezos?

Reply:
It's best if you can get it from RPC. The table would be second best.